src/components/ServiceEndpoints/ServiceEndpoint.tsx

```diff
@@ -1,54 +1,93 @@
-import type { DidUri } from '@kiltprotocol/sdk-js';
+import {
+  Credential,
+  DidUri,
+  ICredential,
+  KiltPublishedCredentialCollectionV1,
+  KiltPublishedCredentialCollectionV1Type,
+} from '@kiltprotocol/sdk-js';
 
-import { useCallback, useState } from 'react';
+import { Fragment, useCallback, useState } from 'react';
 import classnames from 'classnames';
 
+import { map } from 'lodash-es';
+
 import * as styles from './ServiceEndpoint.module.css';
 
 import { CredentialVerifier } from '../Credential/Credential';
 import { useBooleanState } from '../../hooks/useBooleanState';
 
+function isLegacyCredential(credential: unknown): credential is {
+  request: ICredential;
+} {
+  return (
+    typeof credential === 'object' &&
+    credential !== null &&
+    'request' in credential
+  );
+}
+
+function isPublishedCollection(
+  json: unknown,
+  endpointType: string,
+): json is KiltPublishedCredentialCollectionV1 {
+  if (endpointType !== KiltPublishedCredentialCollectionV1Type) {
+    return false;
+  }
+  if (!Array.isArray(json)) {
+    return false;
+  }
+  const [{ credential }] = json as KiltPublishedCredentialCollectionV1;
+  return Credential.isICredential(credential);
```

Comment:
Should this check other array items as well?

Reply:
Oh good point, we were only checking the first credential before